extensions/src/main/java/org/pentaho/platform/plugin/action/ActionRunner.java

```diff
@@ -68,14 +70,17 @@ public ActionRunner( final IAction actionBean, final String actionUser, final Ma
   }
 
   public Boolean call() throws ActionInvocationException {
+    final String workItemUid = ActionUtil.extractUid( params );
     try {
-      return callImpl();
+      final boolean result = callImpl();
+      WorkItemLifecyclePublisher.publish( workItemUid, params, WorkItemLifecyclePhase.SUCCEEDED );
```

Comment:
Agree with @pentaho-esager 's assessment. OOS for this story and therefore for this PR. No further questions here, PR looks good to me